new layout and changes last years session (#6)
vibbits/nextflow-workshop: README.md

Fast prototyping => Custom DSL that enables tasks composition, simplifies most use cases + general purpose programming language for corner cases Easy parallelisation => declarative reactive programming model based on dataflow paradigm, implicit portable parallelism Decouple components => functional approach a task execution is idempotent, ie cannot modify the state of other tasks + isolate dependencies with containers Portable deployments => executor abstraction layer + deployment configuration from implementation logic) 
-->

### Main abstractions
Nextflow consists of four main components: channels, operators, processes and workflows.

- *Channels*: contain the input of the workflows used by the processes. Channels connect processes with each other.
- *Operators*: transform the content of channels by applying functions or transformations. Usually operators are applied on channels to get the input of a process in the right format.
- *Processes*: define the piece of script that is actually being run (e.g. an alignment process with STAR).

> - *Workflows*: call the processes as functions with channels as input arguments, only processes defined in the workflow are run. Workflows were introduced in DSL2.

Reviewer: ```suggestion
- *Workflows*: call the processes as functions with channels as input arguments, only processes defined in the workflow are run.
```

    - Queue channels are designed for connecting the output of one process to the input of other processes.

```
# Value Channels
value_channel1 = Channel.value(1)
value_channel2 = Channel.value("Hello World")
value_channel3 = Channel.value(["a", "b", "c"])

# Queue Channels
queue_channel1 = Channel.of('This', 'is', 'a', 'channel')
queue_channel2 = Channel.fromPath('/path/to/files/*.txt')

```
More info about value and queue channels can be found in the [documentation](https://www.nextflow.io/docs/latest/channel.html#channel-types).

#### 2. Operators
Operators are necessary to transform the content of channels in a format that is necessary for usage in the processes. There is a plethora of different operators[[5](https://www.nextflow.io/docs/latest/operator.html?highlight=view#)], however only a handful are used extensively. Here are some examples that you might come accross:

- `collect`: e.g. when using a channel consisting of multiple independent files (e.g. fastq-files) and need to be assembled for a next process (output in a list data-type).

<details class="admonition admonition-info">

<summary>Note</summary>

> The nextflow documentation details whether the out for each operator is a queue channel or a value channel.

Reviewer 2: ```suggestion
The nextflow documentation details whether each operator produces a queue channel or a value channel.
```
